src/Kafka/CreateKafkaTopic/CreateTopicPage.tsx

```diff
@@ -13,6 +12,8 @@ export type CreateTopicPageProps = {
   checkTopicName: (value: string) => Promise<boolean>;
   availablePartitionLimit: number;
   constantValues: ConstantValues;
+  isSwitchChecked: boolean;
+  setIsSwitchChecked: (value: boolean) => void;
```

Comment:
```suggestion
  onSwitchChange: (value: boolean) => void;
```

Reply:
Names should describe what is happening, not what should happen. That function is a callback for when the checkbox changes its value. The user of the component will decide how to handle that.